packages/flutter_tools/lib/src/fuchsia/fuchsia_kernel_compiler.dart

```diff
@@ -58,35 +58,30 @@ class FuchsiaKernelCompiler {
       '--filesystem-root', fsRoot,
       '--packages', '$multiRootScheme:///$relativePackagesFile',
       '--output', fs.path.join(outDir, '$appName.dil'),
-      // TODO(zra): Add back when this is supported again.
-      // See: https://github.com/flutter/flutter/issues/44925
-      // '--no-link-platform',
-      '--split-output-by-packages',
-      '--manifest', manifestPath,
       '--component-name', appName,
-    ];
 
-    if (buildInfo.isDebug) {
-      flags += <String>[
-        '--embed-sources',
-      ];
-    } else if (buildInfo.isProfile) {
-      flags += <String>[
-        '--no-embed-sources',
-        '-Ddart.vm.profile=true',
-        '--gen-bytecode',
-        '--drop-ast',
-      ];
-    } else if (buildInfo.isRelease) {
-      flags += <String>[
-        '--no-embed-sources',
-        '-Ddart.vm.release=true',
+      // AOT/JIT:
+      if (buildInfo.usesAot) ...<String>['--aot', '--tfa'],
+      if (!buildInfo.usesAot) ...<String>[
+        // TODO(zra): Add back when this is supported again.
+        // See: https://github.com/flutter/flutter/issues/44925
+        // '--no-link-platform',
+        '--split-output-by-packages',
+        '--manifest', manifestPath
+      ],
+
+      // debug, profile, jit release, release:
+      if (buildInfo.isDebug) '--embed-sources',
```

Comment:
This two lines might be better expressed as an `if/else` condition

Reply:
Done